wrangler-service/src/main/java/io/cdap/wrangler/service/directive/SpecificationUpgradeUtils.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.wrangler.service.directive;
+
+import io.cdap.wrangler.PropertyIds;
+import io.cdap.wrangler.dataset.workspace.Workspace;
+import io.cdap.wrangler.proto.connection.ConnectionType;
+import io.cdap.wrangler.service.bigquery.BigQueryHandler;
+import io.cdap.wrangler.service.database.DatabaseHandler;
+import io.cdap.wrangler.service.gcs.GCSHandler;
+import io.cdap.wrangler.service.kafka.KafkaHandler;
+import io.cdap.wrangler.service.s3.S3Handler;
+import io.cdap.wrangler.service.spanner.SpannerHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Specification util specifically needed for upgrade. Can be used to generate the spec that is known by the
+ * connector/source or generate path. This class is mostly hardcoding and should be removed once we remove the
+ * deprecated handlers.
+ */
+public class SpecificationUpgradeUtils {
+  private static final Map<String, String> CONNECTION_TYPE_MAP;
+  static {
+    Map<String, String> connectionTypes = new HashMap<>();
+    connectionTypes.put(ConnectionType.BIGQUERY.name().toLowerCase(), "BigQuery");
+    connectionTypes.put(ConnectionType.GCS.name().toLowerCase(), "GCS");
+    connectionTypes.put(ConnectionType.SPANNER.name().toLowerCase(), "Spanner");
+    connectionTypes.put(ConnectionType.DATABASE.name().toLowerCase(), "Database");
+    connectionTypes.put(ConnectionType.S3.name().toLowerCase(), "S3");
+    connectionTypes.put(ConnectionType.KAFKA.name().toLowerCase(), "Kafka");
+    CONNECTION_TYPE_MAP = connectionTypes;
+  }
+
+  private SpecificationUpgradeUtils() {
+  }
+
+  /**
+   * Get the connector name for the given connection type.
+   *
+   * @param connectionType the connection type
+   * @return the connector name for the connection type
+   */
+  @Nullable
+  public static String getConnectorName(String connectionType) {
```

Comment:
instead of this method, the CONNECTION_TYPE_MAP, and and static initialization block, you can add another property to the ConnectionType enum for the corresponding connector name.